### PR TITLE
[controller] Add ability for selecting bluetooth adapter on linux

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem
We cannot select bluetooth adapter after we use the native bluetooth stack.

 #### Summary of Changes
Add the ability for selecting bluetooth adapter on linux.

### CODE REVIEW

File: src/controller/CHIPDeviceController.cpp [hunk 1]
```diff
@@ -94,7 +94,7 @@ DeviceController::DeviceController()
 }
 
 CHIP_ERROR DeviceController::Init(NodeId localDeviceId, PersistentStorageDelegate * storageDelegate, System::Layer * systemLayer,
-                                  Inet::InetLayer * inetLayer)
+                                  Inet::InetLayer * inetLayer, uint32_t bluetoothAdapterId)
```
Comment (andy31415) [inline src/controller/CHIPDeviceController.cpp:97]: I believe we  are adding too many parameters to init - this turns into very coupled code.

Could we instead force a run of ConfigureBle outside devicecontroller init and let that control the adapter id?

**Review by andy31415 — CHANGES_REQUESTED** [review]

File: src/platform/Linux/bluez/Helper.cpp [hunk 2]
```diff
@@ -1547,6 +1547,11 @@ CHIP_ERROR InitBluezBleLayer(bool aIsCentral, char * apBleAddr, BLEAdvConfig & a
         err = ConfigureBluezAdv(aBleAdvConfig, endpoint);
         SuccessOrExit(err);
     }
+    else
+    {
+        // TODO: The "node id" means the bluetooth adapter on linux, this is confusing and should have a better name.
```
Comment (andy31415) [inline src/platform/Linux/bluez/Helper.cpp:1552]: this is very confusiong. Could we start with a separate rename that changes mNodeId to mAdapterId ?

Comment (erjiaqing) [reply src/platform/Linux/bluez/Helper.cpp]: https://github.com/project-chip/connectedhomeip/pull/5173

File: src/controller/CHIPDeviceController.cpp (hunk 1 above)
Comment (erjiaqing) [reply src/controller/CHIPDeviceController.cpp]: Added a PreConfigureBle api for this, if it is not called before Init the Init function will call it.

**Comment by erjiaqing** [comment]
Convert this to draft for #5173

File: src/controller/CHIPDeviceController.cpp (hunk 1 above)
Comment (andy31415) [reply src/controller/CHIPDeviceController.cpp]: My complaint is that Controller has too much responsibility/functionality. I am questioning why the API to configure BLE is within the controller. I would rather expect it to be decoupled.

Controller can error out if the required subsystems are not configured correctly (e.g. cannot listen on various ports) however I find an extra indirection for BLE configuration to go through a controller to be odd.

File: src/controller/CHIPDeviceController.cpp [hunk 3]
```diff
@@ -104,18 +105,18 @@ CHIP_ERROR DeviceController::Init(NodeId localDeviceId, PersistentStorageDelegat
...
     {
-        mSystemLayer = systemLayer;
-        mInetLayer   = inetLayer;
+        mSystemLayer          = systemLayer;
+        mInetLayer            = inetLayer;
+        mBleManagerConfigured = true;
     }
     else
     {
 #if CONFIG_DEVICE_LAYER
```
Comment (andy31415) [inline src/controller/CHIPDeviceController.cpp:114]: I think this entire 'Init chip stack and ble' has no place inside the controller. It was already bad, we are making it worse by adding more functionality.

Let native callers have a stack initialization - they are fully capable of doing so.

File: src/controller/CHIPDeviceController.cpp (hunk 1 above)
Comment (erjiaqing) [reply src/controller/CHIPDeviceController.cpp]: Moved the BLE init out of the controller.

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @jelderton ?

**Review by saurabhst — APPROVED** [review]

**Pull request merged**